### PR TITLE
feat(skills,ci): encode Phase 1 research cadence patterns + PR review polling script (Closes #507)

### LINKED ISSUE #507: Encode Phase 1 research-to-replanning cadence as reusable skill patterns
# Encode Phase 1 Research-to-Replanning Cadence as Reusable Skill Patterns

## Context

Sprint feat/mcp-web-dashboard Phase 1 (2026-03-28) executed a 10-step model-swap research cadence that produced measurably better results than prior monolithic research delegations. A post-phase retrospective identified 6 encoding gaps — patterns discovered interactively that are not yet in any SKILL.md, agent file, or guide.

These patterns should be encoded before the next research-heavy sprint to prevent re-discovery.

### Gap 1 — Model-Swap Cadence (NEW — deep-research-sprint SKILL.md)
**Target**: `.github/skills/deep-research-sprint/SKILL.md`  
**Section**: New section: "Model-Swap Cadence"  
**Change**: Add 10-step cadence as canonical procedure — Sonnet 4.5 for scouting/execution, Sonnet 4.6 High Reasoning for synthesis/gap-detection/planning; alternating with deliberate human pause gates between substeps; match task type to model capability (deterministic, not ad-hoc)

**Evidence**: Phase 1A–1J cadence produced higher directional alignment and human trust vs. prior single-delegation research sprints. Every gate was a natural correction surface.

### Gap 2 — Skip-Step Optimization (deep-research-sprint SKILL.md)
**Target**: `.github/skills/deep-research-sprint/SKILL.md`  
**Section**: Phase 1F / Gap Triage decision node  
**Change**: Add skip rule — if Gap Triage returns all gaps as CLOSEABLE or DEFERRED_TO_IMPL, skip the tertiary scouting step entirely; log `Phase 1F SKIPPED` in scratchpad with reason; saves ~8,000–12,000 tokens (one Scout context)

**Evidence**: Phase 1E triage closed all 5 open questions; Phase 1F skipped; saved one full model context.

### Gap 3 — Commit Before Replan (deep-research-sprint SKILL.md)
**Target**: `.github/skills/deep-research-sprint/SKILL.md`  
**Section**: Phase ordering constraints  
**Change**: Add constraint — D4 research docs must be committed with `status: Final` before the sprint replanning phase begins. Replanning reads locked final text, enabling precise technology citations in Phase 2+ deliverables. Replanning before commit produces vague "as described in research" references.

**Evidence**: Phase 1G committed both D4 docs (e71ba8f) before Phase 1I replanning; Phase 1I planner cited exact locked decisions from the committed docs by name in all 8 amendments.

### Gap 4 — Executive Planner Input Format (executive-planner.agent.md)
**Target**: `.github/agents/executive-planner.agent.md`  
**Section**: Delegation Input Format section  
**Change**: Add amendment-list format specification — when delegating workplan amendment tasks, provide: (a) a numbered list of exact locked decisions, and (b) a concrete format mandate: `[Phase N — Section/File]: ADD/CHANGE/CLARIFY "exact proposed text" (reason: ...)`

**Evidence**: Phase 1I produced 8 high-precision, non-overlapping amendments in a single delegation. A vague "update the workplan to reflect research findings" prompt would have produced structural prose rather than surgical per-deliverable changes.

### Gap 6 — Phase 1H Issue-Seeding Step (deep-research-sprint SKILL.md)
**Target**: `.github/skills/deep-research-sprint/SKILL.md`  
**Section**: Post-synthesis mandatory steps  
**Change**: Add explicit "Seed deferred-scope issues" step — for each recommendation with `status: deferred` or listed in a Deferred Scope table, create a GitHub issue using `--body-file` pattern (pre-use validation required), record the issue number in the workplan Deferred Scope table, and log the issue numbers in the scratchpad. This step implements the Research Doc PR Merge Gate (AGENTS.md) at the workplan level rather than only at PR level.

**Evidence**: Phase 1H was executed but not named in the skill as a mandatory step. Without it, deferred recommendations from D4 docs would only surface at PR review (AGENTS.md merge gate), not during sprint planning.

### Gap 8 — Library Prohibition Naming (workplan-scaffold SKILL.md)
**Target**: `.github/skills/workplan-scaffold/SKILL.md`  
**Section**: Deliverable specification conventions  
**Change**: Add convention — when a technology choice has a prominent prohibited alternative (heavier library, deprecated tool, framework mismatch), explicitly name the prohibition alongside the chosen technology in the deliverable spec. Format: `use X (do NOT install Y — reason)`

**Evidence**: Phase 4 deliverables named `LayerCake (do NOT install svelte-chartjs or chart.js — 56 KB overhead)` after Phase 1I amendment. Without the explicit prohibition, implementers default to familiar library. Abstract "use LayerCake" is insufficient guardrail.

## Acceptance Criteria

- [ ] Gap 1: `deep-research-sprint/SKILL.md` has a "Model-Swap Cadence" section documenting the alternating 4.5/4.6-HR pattern with pause gate protocol
- [ ] Gap 2: `deep-research-sprint/SKILL.md` has a skip-step decision rule with token-saving rationale
- [ ] Gap 3: `deep-research-sprint/SKILL.md` has an explicit ordering constraint: commit D4 docs before replanning begins
- [ ] Gap 4: `executive-planner.agent.md` includes an amendment-list format spec with the 6-field format example
- [ ] Gap 6: `deep-research-sprint/SKILL.md` includes "Seed deferred-scope issues" as a mandatory post-synthesis step with `--body-file` pattern
- [ ] Gap 8: `workplan-scaffold/SKILL.md` includes library prohibition naming convention with `use X (do NOT install Y)` format example
- [ ] All changes committed and validated by Review agent
- [ ] CI passes on commit

## Routing

| Gap | Target Agent | Target File |
|-----|-------------|------------|
| 1, 2, 3, 6 | Executive Fleet | `.github/skills/deep-research-sprint/SKILL.md` |
| 4 | Executive Fleet | `.github/agents/executive-planner.agent.md` |
| 8 | Executive Docs | `.github/skills/workplan-scaffold/SKILL.md` |

## References

- Scratchpad: `.tmp/feat-mcp-web-dashboard/2026-03-28.md` § Retrospective — 2026-03-28
- Workplan: `docs/plans/2026-03-28-mcp-web-dashboard.md`
- Research docs: `docs/research/mcp-inspector-landscape.md`, `docs/research/svelte-ecosystem-for-webmcp.md`
- Governing skill: `.github/skills/session-retrospective/SKILL.md`

### PR DESCRIPTION
**Pull request by AccessiT3ch**
## Summary

Encodes 6 skill-encoding gaps discovered during the Sprint feat/mcp-web-dashboard Phase 1 (2026-03-28) retrospective. These patterns were identified interactively and are being committed before the next research-heavy sprint to prevent re-discovery.

Also introduces `scripts/wait_for_pr_review.py` — a new polling script for waiting until GitHub PR reviews land — plus provenance report improvements (risk table filtering, agent drift assessment table).

### `.github/skills/deep-research-sprint/SKILL.md` (Gaps 1, 2, 3, 6)

**Gap 1 — Model-Swap Cadence (new section §1.5)**
Adds the alternating Sonnet 4.5 / Sonnet 4.6 High Reasoning cadence as a canonical procedure. Sonnet 4.5 for scouting/execution steps; Sonnet 4.6 HR for synthesis/gap-detection/planning. Includes pause gate protocol and a deterministic model-selection table.

**Gap 2 — Skip-Step Optimisation (§3.5)**
Adds skip rule: if Gap Triage returns all gaps as `CLOSEABLE` or `DEFERRED_TO_IMPL`, skip the tertiary scouting step entirely. Saves ~8,000–12,000 tokens per skipped step. Requires scratchpad log entry with reason.

**Gap 3 — Commit Before Replan (§3.5 Phase Ordering Constraints)**
Adds ordering constraint: D4 research docs must be committed with `status: Final` before sprint replanning begins. Includes canonical example (Phase 1G/1I), anti-pattern description, and required sequence.

**Gap 6 — Seed Deferred-Scope Issues (§5.3.5)**
Adds mandatory post-synthesis step: for every `status: deferred` recommendation, create a GitHub issue using `--body-file` pattern (with pre-use validation), record the issue number in the workplan Deferred Scope table, and log issue numbers in the scratchpad.

### `.github/agents/executive-planner.agent.md` (Gap 4)

**Gap 4 — Amendment-List Input Format (new §5)**
Adds amendment-list format specification for workplan amendment delegations. Provides: (a) numbered locked-decisions input, (b) per-deliverable `[Phase N — Section]: ADD/CHANGE/CLARIFY "exact text" (reason: ...)` format. Includes canonical example from Phase 1I that produced 8 high-precision non-overlapping amendments.

### `.github/skills/workplan-scaffold/SKILL.md` (Gap 8)

**Gap 8 — Library Prohibition Naming (§3.1)**
Adds deliverable specification convention: when a technology choice has a prominent prohibited alternative, explicitly name the prohibition with `use X (do NOT install Y — reason)` format. Canonical example from Phase 4 deliverables.

### `scripts/wait_for_pr_review.py` (new script)

New polling script encoding the PR-review-wait pattern from AGENTS.md. Polls `gh pr view --json reviewRequests,reviews` until all requested reviewers have reviewed (or `--min-reviews` threshold is met in fallback mode). Exit codes: 0 = threshold met, 1 = timeout, 2 = PR not found. Includes 17 tests in `tests/test_wait_for_pr_review.py`.

### `scripts/parse_audit_result.py` + `.github/workflows/audit-provenance.yml`

Provenance report improvements:
- Agent Risk Assessment table now suppresses green agents (shows count summary instead)
- New Agent Drift Assessment table shows before/after risk scores for `.agent.md` files modified in the PR
- Workflow wires `--changed-files` and `--base-sha` from PR context

### `AGENTS.md` + `scripts/README.md`

Updated canonical example and README entry for `wait_for_pr_review.py` with accurate exit code descriptions and `--min-reviews` documentation.

## Closes

Closes #507

## Acceptance Criteria

- [x] Gap 1: `deep-research-sprint/SKILL.md` has a "Model-Swap Cadence" section documenting the alternating 4.5/4.6-HR pattern with pause gate protocol
- [x] Gap 2: `deep-research-sprint/SKILL.md` has a skip-step decision rule with token-saving rationale
- [x] Gap 3: `deep-research-sprint/SKILL.md` has an explicit ordering constraint: commit D4 docs before replanning begins
- [x] Gap 4: `executive-planner.agent.md` includes an amendment-list format spec with the format example
- [x] Gap 6: `deep-research-sprint/SKILL.md` includes "Seed deferred-scope issues" as a mandatory post-synthesis step with `--body-file` pattern
- [x] Gap 8: `workplan-scaffold/SKILL.md` includes library prohibition naming convention with `use X (do NOT install Y)` format example
- [x] `wait_for_pr_review.py` script created with 17 passing tests; polls `reviewRequests` denominator; `--min-reviews` fallback for edge cases
- [x] Provenance report: risk table filters to yellow/red only; drift table added
- [x] All pre-commit hooks pass (validate-agent-files, validate-skill-files, value-encoding-drift, link checks)

### CODE REVIEW

File: .github/skills/deep-research-sprint/SKILL.md [hunk 1]
```diff
@@ -243,6 +308,30 @@ CI must pass before requesting review. Common failure modes after adding a resea
 - Missing source stubs in `docs/research/sources/` → create the stub file
 - `validate_synthesis.py` failure → fix frontmatter or headings
 
+### 5.3.5 Seed Deferred-Scope Issues (Mandatory Post-Synthesis Step)
+
+For every recommendation in the synthesis doc with `status: deferred` in the `recommendations:` frontmatter block, or listed in a `## Deferred Scope` table in the doc body, create a GitHub issue **before the session closes**.
+
+This step implements the [Research Doc PR Merge Gate](../../../AGENTS.md#pr-review-triage-gate) at the workplan level — deferred items are tracked as issues during sprint planning, not only discovered at PR review.
```
[the remark this answers is not in the document]
Comment (AccessiT3ch) [reply .github/skills/deep-research-sprint/SKILL.md]: Fixed in 5b6f6e8 — removed the broken `#pr-review-triage-gate` fragment; link now points to `AGENTS.md` directly (the Research Doc PR Merge Gate has no named anchor in that file).

File: scripts/wait_for_pr_review.py [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+            consecutive_errors = 0
+            current_pending = state["pending"]
+            current_qualifying = _qualifying_reviews(state["reviews"], args.min_body_len, args.states)
+            new_reviews = [r for r in current_qualifying if r["id"] not in baseline_review_ids]
+            new_count = len(new_reviews)
+            latest_new_count = new_count
+            pending_count = len(current_pending)
+
+            print(f"  [{new_count}/{denominator}] {new_count} new review(s) landed, {pending_count} still pending")
+
```
[the remark this answers is not in the document]
Comment (AccessiT3ch) [reply scripts/wait_for_pr_review.py]: Fixed in eceb148 — non-fallback progress line now uses `display_count = min(new_count, denominator)` to cap the fraction at 100%; fallback mode tracks `total_so_far` (total qualifying reviews including baseline) so the fraction is always `[total/threshold]` with no possibility of overflow.
